azerothcore/azerothcore-wotlk
# feat(Core/Misc): Implemented ObjectGuid class. Port from TrinityCore.

## Issues Addressed:
- Closes #3733.



## How to Test the Changes:

This is a very complex PR, so there is nothing specifically to be tested. Just log in, do your stuff (killing mobs, questing, exploring, PvPing, etc) and see if any bugs related to the PR will show up.
Especially, you should keep an eye on loading from/saving to db many things (pets, auras, cooldowns, auctions, mails, mob respawns, etc) and observe errors that can appear in logs/console.

**Connected PRs:**
https://github.com/azerothcore/mod-premium-lib/pull/4
https://github.com/azerothcore/mod-weapon-visual/pull/18
https://github.com/azerothcore/mod-npc-codebox/pull/16
https://github.com/azerothcore/mod-npc-beastmaster/pull/44
https://github.com/azerothcore/mod-morphsummon/pull/13
https://github.com/azerothcore/mod-buff-command/pull/19
https://github.com/azerothcore/mod-transmog/pull/44

**Eluna: https://github.com/ElunaLuaEngine/Eluna/pull/357**

**Is a requirement for:**
#5223

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
